### PR TITLE
Dyn 2560 models folder3

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Increase the code coverage adding test cases for the DynamoCore/Models folder **third** part.
I created several unit tests using nunit and AxoCover (code coverage) for the next classes:

- DynamoModel
- RunSettings

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 

### FYIs

@alfredo-pozo

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
This pull request contains just test methods for the classes inside the DynamoModelCommands.cs and RunSettings.cs files.
I got more than 95.4% of the coverage (using AxoCover) in the Models folder.

image

**Comment by RobertGlobant20** [comment]
This is a screenshot after running all the test cases in the DynamoCoreTest project in Visual Studio, I got the usual 67 cases failed and 2223 passed. Also I ran the DynamoSelfServe job with my branch in master-5 and it finished successfully.

image

File: test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+            Guid newNodeGuid = Guid.NewGuid();
+            //This guid belongs to a CoreNodeModels.CreateList node inside the .dyn file
+            Guid existingNodeGuid = new Guid("81c94fd0-35a0-4680-8535-00aff41192d3");
+            double x = 100;
+            double y = 190;
+
+            string openPath = Path.Combine(TestDirectory, @"core\DetailedPreviewMargin_Test.dyn");
+            RunModel(openPath);
+
+            //This command is created with CreateAsDownstreamNode flas as true
```
Comment (QilongTang) [inline test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs:58]: `flag`

File: test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+            var cmdOne = new DynamoModel.CreateAndConnectNodeCommand(newNodeGuid, existingNodeGuid,
+                "CoreNodeModels.CreateList", 0, 0, x, y, true, false);
+
+            //This command is created with CreateAsDownstreamNode flas as false
```
Comment (QilongTang) [inline test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs:62]: same here

File: test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        public void GetNodeFromCommand_DuplicatedGuidsTest()
+        {
+            //Arrange
+            //This guid belongs to a CoreNodeModels.CreateList node inside the .dyn file
+            Guid existingNodeGuid = new Guid("81c94fd0-35a0-4680-8535-00aff41192d3");
+
+            string openPath = Path.Combine(TestDirectory, @"core\DetailedPreviewMargin_Test.dyn");
+            RunModel(openPath);
+
+            var command = new DynamoModel.CreateNodeCommand(existingNodeGuid.ToString(), "List.Create", 0, 0, true, true);
```
Comment (QilongTang) [inline test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs:116]: Is command `GetNodeFromCommand` part of `CreateNodeCommand`?

File: test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs (hunk 1 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs]: @QilongTang 
Comment fixed in the commit: https://github.com/DynamoDS/Dynamo/pull/10657/commits/e1fa29e30eef851af322f92301c437230be5ff30

File: test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs (hunk 2 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs]: @QilongTang 
Comment fixed in the commit: https://github.com/DynamoDS/Dynamo/pull/10657/commits/e1fa29e30eef851af322f92301c437230be5ff30

File: test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs (hunk 3 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Models/DynamoModelCommandsTest.cs:117]: @QilongTang 
The GetNodeFromCommand() method is called when the CreateNodeCommand is executed, then i added a explanatory comment about it in commit: https://github.com/DynamoDS/Dynamo/pull/10657/commits/e1fa29e30eef851af322f92301c437230be5ff30

File: test/DynamoCoreTests/Models/RunSettingsTest.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        /// <summary>
+        /// This test method will execute the RunSettings constructor without parameters
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void RunSettingsConstructorTest()
+        {
+            //Act
+            //Ejecuting the constructor without parameters
```
Comment (QilongTang) [inline test/DynamoCoreTests/Models/RunSettingsTest.cs:17]: `Executing`

Comment (RobertGlobant20) [reply test/DynamoCoreTests/Models/RunSettingsTest.cs]: @QilongTang 
I fixed spelling in the commit: https://github.com/DynamoDS/Dynamo/pull/10657/commits/e4cb6087f2d7f55cffab39b432839f630fe10bea

**Comment by QilongTang** [comment]
@RobertGlobant20 The selfServce failed for some reason?

**Comment by RobertGlobant20** [comment]
@QilongTang 
> @RobertGlobant20 The selfServce failed for some reason?

If you are talking about the DynamoSelfServe in master-5, yesterday we were seeing issues about docker and also Bruno saw that some test failed but later he generated again the job again and passed successfully, but when I executed the job for my branch finished but says is unstable.

Also I can see that failed in the pullRequestValidation Job due to Docker, log says "Docker container instance failed to initialize".

I will check the pull request problem with Alfredo and also i will launch again the DynamoSelfServe in master-5 just to verify that it runs successfully.

Thanks

**Comment by RobertGlobant20** [comment]
> @RobertGlobant20 The selfServce failed for some reason?

@QilongTang 
I've run again the DynamoSelfServe in master-5 and it was successful (finished without errors).
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/master/80/

Thanks

**Comment by QilongTang** [comment]
@RobertGlobant20 Thanks, merging

**Pull request merged**